system/greenhouse-ccloud/templates/thanos-pluginpreset.yaml

```
@@ -5,9 +5,17 @@ metadata:
   name: thanos
   namespace: ccloud
 spec:
-  {{- if .Values.thanos.clusterOptionOverrides }}
   clusterOptionOverrides:
-    {{ toYaml .Values.thanos.clusterOptionOverrides | nindent 4 }}
+  {{- range $cluster, $ingressPlugin := .Values.ingressPlugins }}
+  {{- $ingressDomain := trimSuffix "." (trimPrefix "ingress." $ingressPlugin.recordName) }}
+  - clusterName: {{ $cluster }}
+    overrides:
+    - name: thanos.query.ingress.grpc.hosts
+      value:
+        - host: "thanos-grpc.{{ $ingressDomain }}"
```

It's a range over a list. So this should work like this:
```suggestion
  {{- range $plugin := .Values.ingressPlugins }}
  {{- $ingressDomain := trimSuffix "." (trimPrefix "ingress." $plugin.recordName) }}
  - clusterName: {{ $plugin.cluster }}
    overrides:
    - name: thanos.query.ingress.grpc.hosts
      value:
        - host: "thanos-grpc.{{ $ingressDomain }}"
```

---

indeed, thanks